engine/source/coupling/rad2rad/r2r_init.F

```diff
@@ -236,8 +247,34 @@ SUBROUTINE R2R_INIT(IEXLNK  ,ITAB,IGRNOD,X      ,
 	ENDIF        
 C----------------------------------------------------------------------              	
 C------	
-        CALL INIT_LINK_C(IDP,NNG,ITAB,GRNOD,X,ADDCNEL,CNEL,IXC,
+        IF (NLLNK(IEX)==1) THEN
+C-------- Coupling of non local dof
+          IDXI => NLOC_DMG%IDXI(1:NUMNOD)
+          POSI => NLOC_DMG%POSI(1:NLOC_DMG%NNOD)
+          COMPT = 0
+          MY_ALLOCATE(NDOF_NL,NNG)
+          DO I=1,NNG
+            NN = IDXI(GRNOD(I))
+            NDOF_NL(I) = POSI(NN+1)-POSI(NN)
+            COMPT = COMPT + NDOF_NL(I)
+          ENDDO
+          NBDOF_NL(IEX) = COMPT
+          ALLOCATE(IADD_NL(NBDOF_NL(IEX)))
```

Comment:
@bertrand-altair

 IADD_NL is allocated here but never deallocated

Reply:
fixed